<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>SaharaWindow</class>
 <widget class="QMainWindow" name="SaharaWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>884</width>
    <height>662</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>OpenPST</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <widget class="QComboBox" name="portList">
    <property name="geometry">
     <rect>
      <x>60</x>
      <y>10</y>
      <width>301</width>
      <height>29</height>
     </rect>
    </property>
    <property name="editable">
     <bool>false</bool>
    </property>
   </widget>
   <widget class="QLabel" name="portListLabel">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>6</y>
      <width>64</width>
      <height>31</height>
     </rect>
    </property>
    <property name="text">
     <string>COM</string>
    </property>
   </widget>
   <widget class="QPlainTextEdit" name="log">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>355</y>
      <width>861</width>
      <height>210</height>
     </rect>
    </property>
   </widget>
   <widget class="QLabel" name="logLabel">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>330</y>
      <width>101</width>
      <height>17</height>
     </rect>
    </property>
    <property name="text">
     <string>Operation Log</string>
    </property>
   </widget>
   <widget class="QPushButton" name="saveLogButton">
    <property name="geometry">
     <rect>
      <x>780</x>
      <y>570</y>
      <width>91</width>
      <height>27</height>
     </rect>
    </property>
    <property name="text">
     <string>Save Log</string>
    </property>
   </widget>
   <widget class="QLabel" name="programNameText">
    <property name="geometry">
     <rect>
      <x>650</x>
      <y>10</y>
      <width>201</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>18</pointsize>
     </font>
    </property>
    <property name="text">
     <string>OpenPST - Sahara</string>
    </property>
   </widget>
   <widget class="QPushButton" name="portRefreshButton">
    <property name="geometry">
     <rect>
      <x>370</x>
      <y>10</y>
      <width>61</width>
      <height>27</height>
     </rect>
    </property>
    <property name="text">
     <string>Refresh</string>
    </property>
   </widget>
   <widget class="QPushButton" name="portDisconnectButton">
    <property name="enabled">
     <bool>true</bool>
    </property>
    <property name="geometry">
     <rect>
      <x>520</x>
      <y>10</y>
      <width>91</width>
      <height>25</height>
     </rect>
    </property>
    <property name="text">
     <string>Disconnect</string>
    </property>
   </widget>
   <widget class="QTabWidget" name="saharaTabSet">
    <property name="enabled">
     <bool>true</bool>
    </property>
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>60</y>
      <width>861</width>
      <height>251</height>
     </rect>
    </property>
    <property name="currentIndex">
     <number>1</number>
    </property>
    <property name="documentMode">
     <bool>false</bool>
    </property>
    <widget class="QWidget" name="saharaXmlTab">
     <attribute name="title">
      <string>Provision XML</string>
     </attribute>
     <widget class="QLineEdit" name="saharaXmlFilePath">
      <property name="enabled">
       <bool>false</bool>
      </property>
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>30</y>
        <width>381</width>
        <height>25</height>
       </rect>
      </property>
     </widget>
     <widget class="QToolButton" name="saharaXmlFileBrowseButton">
      <property name="enabled">
       <bool>false</bool>
      </property>
      <property name="geometry">
       <rect>
        <x>400</x>
        <y>30</y>
        <width>25</width>
        <height>23</height>
       </rect>
      </property>
      <property name="text">
       <string>...</string>
      </property>
     </widget>
     <widget class="QLabel" name="label">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>10</y>
        <width>61</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>XML File</string>
      </property>
     </widget>
     <widget class="QPushButton" name="saharaXmlProvisionButton">
      <property name="enabled">
       <bool>false</bool>
      </property>
      <property name="geometry">
       <rect>
        <x>440</x>
        <y>30</y>
        <width>111</width>
        <height>23</height>
       </rect>
      </property>
      <property name="text">
       <string>Provision Phone</string>
      </property>
     </widget>
    </widget>
    <widget class="QWidget" name="saharaManualTab">
     <attribute name="title">
      <string>Manual Processing</string>
     </attribute>
     <widget class="QPushButton" name="saharaReadHelloButton">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>10</y>
        <width>91</width>
        <height>25</height>
       </rect>
      </property>
      <property name="text">
       <string>Read Hello</string>
      </property>
     </widget>
     <widget class="QPushButton" name="saharaWriteHelloButton">
      <property name="geometry">
       <rect>
        <x>200</x>
        <y>40</y>
        <width>91</width>
        <height>25</height>
       </rect>
      </property>
      <property name="text">
       <string>Write Hello</string>
      </property>
     </widget>
     <widget class="QComboBox" name="saharaWriteHelloSwitchCommandSelect">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>40</y>
        <width>181</width>
        <height>25</height>
       </rect>
      </property>
      <property name="editable">
       <bool>false</bool>
      </property>
     </widget>
     <widget class="QLineEdit" name="deviceStateModeValue">
      <property name="enabled">
       <bool>false</bool>
      </property>
      <property name="geometry">
       <rect>
        <x>410</x>
        <y>10</y>
        <width>161</width>
        <height>20</height>
       </rect>
      </property>
      <property name="text">
       <string>Waiting Hello</string>
      </property>
     </widget>
     <widget class="QLabel" name="deviceStateModeLabel">
      <property name="geometry">
       <rect>
        <x>310</x>
        <y>10</y>
        <width>91</width>
        <height>20</height>
       </rect>
      </property>
      <property name="text">
       <string>Current Mode</string>
      </property>
     </widget>
     <widget class="QLabel" name="deviceStateVersionLabel">
      <property name="geometry">
       <rect>
        <x>590</x>
        <y>10</y>
        <width>91</width>
        <height>20</height>
       </rect>
      </property>
      <property name="text">
       <string>Protocol Ver.</string>
      </property>
     </widget>
     <widget class="QLineEdit" name="deviceStateVersionValue">
      <property name="enabled">
       <bool>false</bool>
      </property>
      <property name="geometry">
       <rect>
        <x>680</x>
        <y>10</y>
        <width>161</width>
        <height>20</height>
       </rect>
      </property>
      <property name="text">
       <string>Waiting Hello</string>
      </property>
     </widget>
     <widget class="QLabel" name="deviceStateMinVersionLabel">
      <property name="geometry">
       <rect>
        <x>590</x>
        <y>40</y>
        <width>91</width>
        <height>20</height>
       </rect>
      </property>
      <property name="text">
       <string>Min Version</string>
      </property>
     </widget>
     <widget class="QLineEdit" name="deviceStateMinVersionValue">
      <property name="enabled">
       <bool>false</bool>
      </property>
      <property name="geometry">
       <rect>
        <x>680</x>
        <y>40</y>
        <width>161</width>
        <height>20</height>
       </rect>
      </property>
      <property name="text">
       <string>Waiting Hello</string>
      </property>
     </widget>
     <widget class="QLabel" name="deviceStatePreferredMaxSizeLabel">
      <property name="geometry">
       <rect>
        <x>310</x>
        <y>40</y>
        <width>91</width>
        <height>20</height>
       </rect>
      </property>
      <property name="text">
       <string>Pref. Max Size</string>
      </property>
     </widget>
     <widget class="QLineEdit" name="deviceStatePreferredMaxSizeValue">
      <property name="enabled">
       <bool>false</bool>
      </property>
      <property name="geometry">
       <rect>
        <x>410</x>
        <y>40</y>
        <width>161</width>
        <height>20</height>
       </rect>
      </property>
      <property name="text">
       <string>Waiting Hello</string>
      </property>
     </widget>
     <widget class="QPushButton" name="saharaSwitchModeButton">
      <property name="geometry">
       <rect>
        <x>230</x>
        <y>120</y>
        <width>121</width>
        <height>25</height>
       </rect>
      </property>
      <property name="text">
       <string>Switch Mode</string>
      </property>
     </widget>
     <widget class="QComboBox" name="saharaSwitchModeSelect">
      <property name="enabled">
       <bool>true</bool>
      </property>
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>120</y>
        <width>211</width>
        <height>25</height>
       </rect>
      </property>
      <property name="editable">
       <bool>false</bool>
      </property>
     </widget>
     <widget class="QPushButton" name="saharaSendClientCommandButton">
      <property name="geometry">
       <rect>
        <x>230</x>
        <y>150</y>
        <width>161</width>
        <height>25</height>
       </rect>
      </property>
      <property name="text">
       <string>Send Client Command</string>
      </property>
     </widget>
     <widget class="QComboBox" name="saharaSendClientCommandSelect">
      <property name="enabled">
       <bool>true</bool>
      </property>
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>150</y>
        <width>211</width>
        <height>25</height>
       </rect>
      </property>
      <property name="editable">
       <bool>false</bool>
      </property>
     </widget>
     <widget class="QPushButton" name="saharaSendImageButton">
      <property name="geometry">
       <rect>
        <x>720</x>
        <y>120</y>
        <width>121</width>
        <height>25</height>
       </rect>
      </property>
      <property name="text">
       <string>Send Image</string>
      </property>
     </widget>
     <widget class="QToolButton" name="saharaSendImageBrowse">
      <property name="geometry">
       <rect>
        <x>820</x>
        <y>90</y>
        <width>25</width>
        <height>23</height>
       </rect>
      </property>
      <property name="text">
       <string>...</string>
      </property>
     </widget>
     <widget class="QLineEdit" name="saharaSendImagePath">
      <property name="geometry">
       <rect>
        <x>430</x>
        <y>90</y>
        <width>381</width>
        <height>25</height>
       </rect>
      </property>
     </widget>
     <widget class="QPushButton" name="saharaSendCommandButton">
      <property name="geometry">
       <rect>
        <x>230</x>
        <y>90</y>
        <width>121</width>
        <height>25</height>
       </rect>
      </property>
      <property name="text">
       <string>Send Command</string>
      </property>
     </widget>
     <widget class="QComboBox" name="saharaSendCommandSelect">
      <property name="enabled">
       <bool>true</bool>
      </property>
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>90</y>
        <width>211</width>
        <height>25</height>
       </rect>
      </property>
      <property name="editable">
       <bool>false</bool>
      </property>
     </widget>
    </widget>
    <widget class="QWidget" name="saharaTabStreamingDload">
     <attribute name="title">
      <string>Transition to Streaming DLOAD</string>
     </attribute>
     <widget class="QLineEdit" name="streamingDloadHelloMagiInput">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>30</y>
        <width>261</width>
        <height>27</height>
       </rect>
      </property>
      <property name="text">
       <string>QCOM fast download protocol host</string>
      </property>
     </widget>
     <widget class="QLineEdit" name="streamingDloadHelloVersionInput">
      <property name="geometry">
       <rect>
        <x>280</x>
        <y>30</y>
        <width>113</width>
        <height>27</height>
       </rect>
      </property>
      <property name="text">
       <string>4</string>
      </property>
     </widget>
     <widget class="QLineEdit" name="streamingDloadHelloCompatVersionInput">
      <property name="geometry">
       <rect>
        <x>400</x>
        <y>30</y>
        <width>113</width>
        <height>27</height>
       </rect>
      </property>
      <property name="text">
       <string>4</string>
      </property>
     </widget>
     <widget class="QLabel" name="streamingDloadHelloLabel">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>10</y>
        <width>67</width>
        <height>17</height>
       </rect>
      </property>
      <property name="text">
       <string>Magic</string>
      </property>
     </widget>
     <widget class="QLabel" name="streamingDloadHelloVersionLabel">
      <property name="geometry">
       <rect>
        <x>280</x>
        <y>10</y>
        <width>67</width>
        <height>17</height>
       </rect>
      </property>
      <property name="text">
       <string>Version</string>
      </property>
     </widget>
     <widget class="QLabel" name="streamingDloadHelloFeatureBitsLabel">
      <property name="geometry">
       <rect>
        <x>530</x>
        <y>10</y>
        <width>91</width>
        <height>17</height>
       </rect>
      </property>
      <property name="text">
       <string>Feature Bits</string>
      </property>
     </widget>
     <widget class="QLineEdit" name="streamingDloadHelloFeatureBitsInput">
      <property name="geometry">
       <rect>
        <x>530</x>
        <y>30</y>
        <width>113</width>
        <height>27</height>
       </rect>
      </property>
      <property name="text">
       <string>4</string>
      </property>
     </widget>
     <widget class="QLabel" name="streamingDloadHelloCompatVersionLabel">
      <property name="geometry">
       <rect>
        <x>400</x>
        <y>10</y>
        <width>101</width>
        <height>17</height>
       </rect>
      </property>
      <property name="text">
       <string>Compat. Version</string>
      </property>
     </widget>
     <widget class="QPushButton" name="streamingDloadHelloButton">
      <property name="geometry">
       <rect>
        <x>650</x>
        <y>30</y>
        <width>99</width>
        <height>27</height>
       </rect>
      </property>
      <property name="text">
       <string>Say Hello</string>
      </property>
     </widget>
    </widget>
   </widget>
   <widget class="QPushButton" name="saharaReadPortButton">
    <property name="geometry">
     <rect>
      <x>680</x>
      <y>320</y>
      <width>191</width>
      <height>31</height>
     </rect>
    </property>
    <property name="text">
     <string>Attempt Read (Max Size)</string>
    </property>
   </widget>
   <widget class="QPushButton" name="clearLogButton">
    <property name="geometry">
     <rect>
      <x>680</x>
      <y>570</y>
      <width>91</width>
      <height>27</height>
     </rect>
    </property>
    <property name="text">
     <string>Clear Log</string>
    </property>
   </widget>
   <widget class="QPushButton" name="portConnectButton">
    <property name="geometry">
     <rect>
      <x>440</x>
      <y>10</y>
      <width>71</width>
      <height>25</height>
     </rect>
    </property>
    <property name="text">
     <string>Connect</string>
    </property>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>884</width>
     <height>25</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuQC_Tools">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionClose"/>
   </widget>
   <widget class="QMenu" name="menuHelp">
    <property name="title">
     <string>Help</string>
    </property>
    <addaction name="actionAbout"/>
   </widget>
   <addaction name="menuQC_Tools"/>
   <addaction name="menuHelp"/>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <action name="actionTest">
   <property name="text">
    <string>test</string>
   </property>
  </action>
  <action name="actionAbout">
   <property name="text">
    <string>About</string>
   </property>
  </action>
  <action name="actionClose">
   <property name="text">
    <string>Close</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections>
  <connection>
   <sender>portRefreshButton</sender>
   <signal>clicked()</signal>
   <receiver>SaharaWindow</receiver>
   <slot>update()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>400</x>
     <y>63</y>
    </hint>
    <hint type="destinationlabel">
     <x>441</x>
     <y>333</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
